src/Microsoft.ML.Core/ComponentModel/ComponentCatalog.cs

```diff
@@ -355,7 +355,7 @@ internal sealed class EntryPointInfo
                 var parameters = method.GetParameters();
                 if (parameters.Length != 2 && parameters.Length != 3)
                     throw Contracts.Except("Method '{0}' has {1} parameters, but must have 2 or 3", method.Name, parameters.Length);
-                if (parameters[0].ParameterType != typeof(IHostEnvironment))
+                if (parameters[0].ParameterType != typeof(IHostEnvironment) && parameters[0].ParameterType != typeof(ISeededEnvironment))
```

Comment:
I wouldn't do it this way.

Instead, I would keep `DataOperationsCatalog` taking an `IHostEnvironment` as a parameter, and casting it down to `ISeededEnvironment` when you need to grab the `Seed` property.